frontend/src/views/HomeView.vue

```diff
@@ -1 +1,51 @@
-<template><h1>Test</h1></template>
+<template>
+  <div>
+    <el-button @click="showDialog">Add Knowledge</el-button>
+    <el-dialog v-model="dialogVisible" title="Add Knowledge">
+      <el-input v-model="newKnowledge" placeholder="Enter knowledge name"></el-input>
+      <span slot="footer" class="dialog-footer">
+        <el-button @click="dialogVisible = false">Cancel</el-button>
+        <el-button type="primary" @click="saveKnowledgeAction">Save</el-button>
+      </span>
+    </el-dialog>
+  </div>
+</template>
+
+<script>
+import { ref } from 'vue'
+import { saveKnowledge } from './knowledgeApi'
+
+export default {
+  setup() {
+    const dialogVisible = ref(false)
+    const newKnowledge = ref('')
+
+    const showDialog = () => {
+      dialogVisible.value = true
+    }
+
+    const saveKnowledgeAction = async () => {
+      try {
+        const success = await saveKnowledge(newKnowledge.value)
```

Comment:
You don't have to call any other function than the action function of your state management to do this. Please, import the state, and use the function inside the actions file

Reply:
done